### ISSUE TITLE
Gauge rendering seems to have some shared coordinates

### ISSUE THREAD

**Issue by MuteTiefling** [TeamPneumatic/pnc-repressurized#1363]
### Describe the bug

Hard to describe. Picture works well though 
![image](https://github.com/user-attachments/assets/6094bfe8-06a9-474b-bc64-e45cc4ba5e46)

### How to reproduce the bug

Place 2 or more gauges in proximity

### Expected behavior

The gauge's rendering should be contained to its own instance

### Additional details

This appears to be fully caused by Sodium, but I wanted to report it to you as well, just to keep you in the loop. 

### Which Minecraft version are you using?

1.21

### Which version of PneumaticCraft: Repressurized are you using?

8.2.2

**Comment by desht** [comment]
Yep, bug in Iris. I reported this back in 1.19/1.20 to the Oculus maintainer, but was ignored.

The good news is, this is basically fixed in Monocle already, so when that's available... :wink: 

In the meantime the workaround is to switch the gauges to text-only mode, using the Pneumatic Wrench.

**Comment by theboo** [comment]
I know this is still another mod issue, but I just tested with the recent monocle release, embeddium, and iris 1.8 beta 8(the version monocle currently requires)... Issue persists there.

Sorry for phone pics, tested on a laptop I don't want to be logged in to GitHub.. sodium is disabled 
![PXL_20241117_171647263](https://github.com/user-attachments/assets/d18b325a-a0fd-4467-924d-595bac5eddef)
 
![PXL_20241117_171702275](https://github.com/user-attachments/assets/1092f4e5-0651-487c-acf0-80c080c1ef4a)

**Comment by desht** [comment]
Ok, will mention that to the Monocle/Embeddium guys, it's possible the fix that we were playing around with didn't make it into the release...

**Comment by embeddedt** [comment]
I think by now Ferri has added it. Might be worth testing again with the latest alpha on CF.

**Comment by MuteTiefling** [comment]
Nope, not fixed. 

![Image](https://github.com/user-attachments/assets/ea310e61-0f54-4e47-b80e-70ad75b3571e)

![Image](https://github.com/user-attachments/assets/28a679ab-8e6b-4a70-af95-984ecf482ebc)

![Image](https://github.com/user-attachments/assets/dd38c018-626e-4728-a1e8-32b9bc3975cd)

**Comment by Lolothepro** [comment]
I noticed that Minecraft Vanilla/Iris had a similar bug with fishing rods:
https://report.bugs.mojang.com/servicedesk/customer/portal/2/MC-300305

The issue was fixed in Vanilla Minecraft with the following change:

```
--- 25w31a/src/net/minecraft/client/renderer/entity/FishingHookRenderer.java
+++ 25w32a/src/net/minecraft/client/renderer/entity/FishingHookRenderer.java
@@ -44,11 +44,14 @@
       float var4 = (float)var1.lineOriginOffset.x;
       float var5 = (float)var1.lineOriginOffset.y;
       float var6 = (float)var1.lineOriginOffset.z;
-      var3.submitCustomGeometry(var2, RenderType.lineStrip(), (var3x, var4x) -> {
+      var3.submitCustomGeometry(var2, RenderType.lines(), (var3x, var4x) -> {
          byte var5x = 16;
 
-         for (int var6x = 0; var6x <= 16; var6x++) {
-            stringVertex(var4, var5, var6, var4x, var3x, fraction(var6x, 16), fraction(var6x + 1, 16));
+         for (int var6x = 0; var6x < 16; var6x++) {
+            float var7 = fraction(var6x, 16);
+            float var8 = fraction(var6x + 1, 16);
+            stringVertex(var4, var5, var6, var4x, var3x, var7, var8);
+            stringVertex(var4, var5, var6, var4x, var3x, var8, var7);
          }
       });
       var2.popPose();

```
